testing_dir/minesweeper.py

```diff
@@ -89,7 +89,7 @@ def game_loop():
             player_pos = new_pos
             moves += 1
 
-        if moves == BOARD_SIZE * BOARD_SIZE - NUM_MINES:
+        if moves == BOARD_SIZE * BOARD_SIZE - NUM_MINES + 1:  # Error: should not have +1
```

Comment:
The comment indicates an error in the logic. The condition should not have `+1` because it would prematurely end the game one move early.
```diff
-        if moves == BOARD_SIZE * BOARD_SIZE - NUM_MINES + 1:  # Error: should not have +1
+        if moves == BOARD_SIZE * BOARD_SIZE - NUM_MINES:
```